### ISSUE TITLE
Errors when running locally

### ISSUE THREAD

**Issue by ptrwtts** [reservoirprotocol/sample-marketplace#23]
The goal should be that a user can check out the repo, run it locally, and it works out of the box. I got two different errors when I tried to do this. 

On the root URL:

```
error - lib/params.ts (11:41) @ eval
TypeError: Cannot read properties of undefined (reading 'toString')
   9 | ) {
  10 |   Object.keys(query).map((key) =>
> 11 |     url.searchParams.set(key, query[key].toString())
     |                                         ^
  12 |   )
  13 | }
```

And when using a subdomain, it failed if I did not specify open sea api key. What is this used for? Also why does it say that an infura key is needed? What is that used for?

**Comment by diegoramosxyz** [comment]
I fixed both errors on `dev`. The subdomain will not cause that issue, and it will only ask for an OpenSea API key if they attempt to post to OpenSea without an API key.

With regards to the Infura Provider, [ethers.js](https://docs.ethers.io/v5/api/providers/#providers-getDefaultProvider) explains the following:

> It is highly recommended for production services to acquire and specify an API Key for each service.
The default API Keys used by ethers are shared across all users, so services may throttle all services that are using the default API Keys during periods of load without realizing it.
Many services also have monitoring and usage metrics, which are only available if an API Key is specified. This allows tracking how many requests are being sent and which methods are being used the most.
Some services also provide additional paid features, which are only available when specifying an API Key.

For the purpose of forking, I was planning have a forkable repo and a live repo for our stuff. In that forkable repo  I'm going to use the default API keys from ethers.js and only remind devs about this. I had talked to @rorygeoghegan the idea of a forkable repo vs a live repo because using wildcard introduces a lot of code only useful if using wildcards. Devs would have to remove this code, so my plan is to do that for them in the forkable repo.
